dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/FilteringZipDependencySelector.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.opensource.dependencies;
+
+import java.util.Map;
+import org.eclipse.aether.artifact.Artifact;
+import org.eclipse.aether.collection.DependencyCollectionContext;
+import org.eclipse.aether.collection.DependencySelector;
+import org.eclipse.aether.graph.Dependency;
+
+/** Filters artifacts with {@code zip} type. */
+public class FilteringZipDependencySelector implements DependencySelector {
+  // To exclude log4j-api-java9:zip:2.11.1, which is not published.
```

Comment:
What is the artifact? Why is it packaged as a zip? I don't think zip is a standard maven packaging type. What's going on here?

Reply:
The artifact is not intended to publish to Maven Central; so they can use arbitrary packaging type.

In log4j project, artifact log4j-api-java9 contains Java 9-specific Java source code. The files in the archive are processed by `log4j-api` project and then added to `META-INF/versions/9` in log4j-api's JAR. Detail: https://github.com/GoogleCloudPlatform/cloud-opensource-java/issues/339 .